mayastor/src/replicas/rebuild/rebuild_impl.rs

```diff
@@ -205,6 +213,17 @@ impl RebuildJob {
             &mut self.tasks.buffers[id as usize]
         };
 
+        let len = copy_buffer.len() as u64 / self.block_size;
+        let mut ctx = RangeContext::new(blk, len, self.nexus_channel.clone());
+
+        self.nexus_descriptor
+            .lock_lba_range(&mut ctx)
```

Comment:
If we need to hold on to the context where are going to put it?

Reply:
The ctx is only required for the duration of the copy. Once the unlock is called on line 241 we no longer require it.

Reply:
Right, but where we do we store for the duration of the lock?